kong/cache/init.lua

```diff
@@ -140,13 +140,25 @@ function _M.new(opts)
     neg_ttl        = neg_ttl,
   }
 
-  local ok, err = cluster_events:subscribe("invalidations", function(key)
-    log(DEBUG, "received invalidate event from cluster for key: '", key, "'")
-    self:invalidate_local(key)
-  end)
-  if not ok then
-    return nil, "failed to subscribe to invalidations cluster events " ..
-                "channel: " .. err
+  if self.shm_name == "kong_db_cache" then
```

Comment:
I would prefer inverting the conditions, so that we do not risk mixing the core cache with other caches.

Reply:
Also, maybe instead of evaluating this multiple times (the code that calls `cluster_events:broadcast()` also contains the same conditional expression), we could just evaluate it once when creating a cache instance and stash it as `self.invalidation_channel` or something.

Reply:
nice idea!

Reply:
I've updated as suggested above